Introduced some other security fixes

filippolauria opened the pull request.

lirantal commented:
@filippolauria thanks for sending it over.
Can you outline the overall changes that this introduces and whether it would be a breaking change for more recent PHP versions or not?

filippolauria commented:
Hi @lirantal ,
sure.
- dologin.php - validating user input and introduced CSRF token check
- logging.php - preventing direct access to this file, which means redirecting to index.php when directly hitting this include file, (repeated for other exten. / include / etc. files), and optimized logging features
- actionMessages.php and errorHandling.php same as logging.php
- library/exten-* files - preventing direct access to these files, generalized the approach of looking for known file locations, improved user input validation (params such as limit, etc.), improved output (escaping logfile output, etc.)
- in library/exten-server_info.php - i also have added new commands for retrieving network information and system distro/version
- opendb.php - general code optimization (TODO prevent direct access to this file)
- sessions.php - introduced new session handling (TODO prevent direct access to this file)
- login.php - removed the default username hardcoded in the form, added the csrf_token
- rep-* - user input validation, code optimization

This is what comes to my mind just by skimming code through github diff interface.

About backward compatibility. For what I can tell, I think the changes I made don't break anything.
Clearly I am testing them - as good as I can - as I implement them on (debian 11, apache 2.4, php 7.4, freeradius 3.0, mariadb 10.5).

lirantal commented:
Thanks! Appreciate the elaborate update. I'll add that as part of the commit message too.
Also, adding you as a contributor to the project. Thank you! :)
